File: testsrc/net/jimmc/mimprint/TestPlayItem.java
package net.jimmc.mimprint;

import net.jimmc.util.SuiteHelper;

import junit.framework.Test;
import junit.framework.TestCase;

import java.io.PrintWriter;
import java.io.StringWriter;

public class TestPlayItem extends TestCase {
    public TestPlayItem(String name) {
    	super(name);
    }

    public static Test suite() {
	Class testClass = TestPlayItem.class;
	Class[] coverageClasses = { PlayItem.class };
	return SuiteHelper.testOrCoverage(testClass,coverageClasses);
    }

    public void testIsImageInfoLine() {
        PlayItem item = new PlayItem();
        assertFalse(item.isImageInfoLine(""));
        assertFalse(item.isImageInfoLine("#comment"));
        assertFalse(item.isImageInfoLine("+command"));
        assertFalse(item.isImageInfoLine("-command"));
        assertTrue(item.isImageInfoLine("filename.jpg"));
        assertTrue(item.isImageInfoLine("filename.gif;+r"));
    }

    public void testAddLine() {
        PlayItem item = new PlayItem();
        assertEquals(0,item.lineCount());
        item.addTextLine("line 1");
        item.addTextLine("line 2");
        assertEquals(2,item.lineCount());
    }

    public void testImageInfoLine() {
        PlayItem item = new PlayItem();
        item.setImageInfoLine("foo");
        assertEquals("foo",item.getImageInfoLine());
        assertEquals("foo",item.getFileName());
        assertEquals(0,item.getRotFlag());

        item = new PlayItem();
        item.setImageInfoLine("foo.gif;;");
        assertEquals("foo.gif",item.getImageInfoLine());
        assertEquals("foo.gif",item.getFileName());
        assertEquals(0,item.getRotFlag());

        item = new PlayItem();
        item.setImageInfoLine("bar;-r");
        assertEquals("bar;-r",item.getImageInfoLine());
        assertEquals("bar",item.getFileName());
        assertEquals(-1,item.getRotFlag());

        item = new PlayItem();
        item.setImageInfoLine("foo.jpg;+r");
        assertEquals("foo.jpg;+r",item.getImageInfoLine());
        assertEquals("foo.jpg",item.getFileName());
        assertEquals(1,item.getRotFlag());
    }

    public void testPrintAll() {
        PlayItem item = new PlayItem();
        item.addTextLine("#line 1");
        item.addTextLine("#line 2");
        item.setImageInfoLine("foo.jpg;-r");
        StringWriter out = new StringWriter();
        item.printAll(new PrintWriter(out));
        String refStr = "#line 1\n#line 2\nfoo.jpg;-r\n";
        assertEquals(refStr,out.toString());
    }

    public void testCopy() {
        PlayItem item = new PlayItem();
        item.addTextLine("#line 1");
        item.addTextLine("#line 2");
        item.setImageInfoLine("foo.jpg;-r");

        PlayItem c = (PlayItem)item.clone();    //calls copy in PlayItem
        assertEquals(2,c.lineCount());
        assertEquals("foo.jpg",c.getFileName());
        assertEquals(-1,c.getRotFlag());
    }

}

File: testsrc/net/jimmc/mimprint/TestPlayItem.scala
package net.jimmc.mimprint

import org.scalatest.Suite

import java.io.PrintWriter
import java.io.StringWriter

class TestPlayItem extends Suite {

    def testIsFinalLine() {
        assert(!PlayItem.isFinalLine(""))
        assert(!PlayItem.isFinalLine("#comment"))
        assert(!PlayItem.isFinalLine("+command"))
        assert(!PlayItem.isFinalLine("-command"))
        assert(PlayItem.isFinalLine("filename.jpg"))
        assert(PlayItem.isFinalLine("filename.jpg"))
        assert(PlayItem.isFinalLine("-empty"));
    }

    def testIsCommentLine() {
        assert(PlayItem.isCommentLine("#comment"))
        assert(!PlayItem.isCommentLine("not a comment"))
    }

    def testIsEmptyLine() {
        assert(PlayItem.isEmptyLine("-empty"))
        assert(!PlayItem.isEmptyLine("foo"))
    }

    def testImageInfoLine() {
        val item1 = new PlayItem(null,null,"foo",0)
        assert("foo"==item1.getImageInfoLine())
        assert("foo"==item1.getFileName())
        assert(0==item1.getRotFlag())

        val item2 = new PlayItem(null,null,"foo.gif",0)
        assert("foo.gif"==item2.getImageInfoLine())
        assert("foo.gif"==item2.getFileName())
        assert(0==item2.getRotFlag())

        val item3 = new PlayItem(null,null,null,0)
        assert("-empty"==item3.getImageInfoLine())
        assert(null==item3.getFileName())
        assert(0==item3.getRotFlag())
    }

    def testIsEmpty() {
        val item1 = new PlayItem(null,null,null,0)
        assert(item1.isEmpty)

        val item2 = new PlayItem(null,null,"foo.gif",0)
        assert(!item2.isEmpty)
    }

    def testApply() {
        val lines = List("#comment1","foo.jpg")
        val item1 = PlayItem(lines, null)
        assert("foo.jpg"==item1.getFileName)
        assert(0==item1.getRotFlag)
    }

/*
    def testPrintAll() {
        val comments = List("#line 1","#line 2")
        val fn = "foo.jpg"
        val rot = -1
        val item = new PlayItem(comments,null,fn,rot)
        val out = new StringWriter()
        item.printAll(new PrintWriter(out), null)
//There is now an extra comment about the base dir
        val refStr = "#line 1\n#line 2\nfoo.jpg;-r\n"
println("test is "+out.toString())
println("refstr is "+resStr)
        assert(refStr==out.toString())
    }
*/
}
